Kconfig

```diff
@@ -200,6 +200,14 @@ config OPTIMIZE_FOR_NONE
 
 endchoice
 
+config OPTIMIZE_FLAG
+	string
+	default "2" if OPTIMIZE_FOR_PERFORMANCE
```

Comment:
Best to set the whole flag here i.e. "O2" instead of "2". Also default should be performance.

Reply:
Default optimization value in correct setup is set to performance in `choice prompt "Optimization"`, so in `config OPTIMIZE_FLAG` also.  Default value in  `config OPTIMIZE_FLAG` is invalid value because when somebody add new optimization value (i.e. -O3) then it should be added in `choice prompt "Optimization"` and in `config OPTIMIZE_FLAG`, so when somebody forget about it there will be compilation error instead of compilation with default flag.

Reply:
Ok, so under what circumstances would we get the compile error ? 
Compile errors through users failing to configure a flag should never happen as default safe values should always be used.

Reply:
When somebody add new optimization level, for example -O3, so will look similar to:
```
choice
	prompt "Optimization"
	default OPTIMIZE_FOR_PERFORMANCE
	help
	  [...]

config OPTIMIZE_FOR_PERFORMANCE_3
	bool "Optimize for performance the most"
	help
	  Apply compiler optimizations prioritizing performance.
	  It means -O3 for GCC or equivalent for other compilers.
[...]
endchoice
``` 
and `OPTIMIZE_FLAGS` variable won't be modified.

In next step, user choose new option from Kconfig - `OPTIMIZE_FOR_PERFORMANCE_3`, then in my solution there will be error during compilation (set flag "OX", so quick diagnose of error). When I make valid default flag in `OPTIMIZE_FLAGS` (for example "O2") there won't be any error/warning message so it will be easy to miss optimization flag used in compiler out. 

Moreover such a situation describe problem with code, not with used configuration, so I think that there should be error message from code (compiler).